build/Build.cs

```diff
@@ -118,8 +118,12 @@ class Build : NukeBuild
                 // 2) that temporary assembly plus all of the third-party assemblies, leaving only the types from the first (Octopus temporary) assembly as public.
                 // --andrewh 14/2/2022.
 
-                // Stage 1: Merge all the Octopus assemblies.
-                var stage1Assemblies = inputFolder.GlobFiles("Octopus.*.dll")
+                // Stage 1: Merge all the Octopus assemblies whose contracts we want to not internalize.
+                var stage1Assemblies = inputFolder.GlobFiles(
+                        "Octopus.Server.Client.dll",
+                        "Octopus.Server.MessageContracts.Base.dll",
+                        "Octopus.Server.MessageContracts.Base.HttpRoutes.dll"
+                    )
```

Comment:
This is the meat of this change: be specific about the assemblies we want to keep public.